great_expectations/datasource/fluent/spark_generic_splitters.py

```diff
@@ -0,0 +1,300 @@
+from __future__ import annotations
```

Comment:
@anthonyburdi How come these splitters are called "generic"?  Are there non-generic ones?  Thanks.

Reply:
Good question! I was differentiating since child datasources might have different splitters / splitter implementations. For example, there are "generic" sql splitters in sql_datasource.py https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/datasource/fluent/sql_datasource.py#L190, and sqlite specific splitters in sqlite_datasource.py, e.g.: https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/datasource/fluent/sqlite_datasource.py#L41